xmtp_mls/src/groups/sync.rs

```diff
@@ -151,6 +151,15 @@ impl MlsGroup {
                     return Ok(());
                 }
                 Ok(Some(intent)) => {
+                    if intent.state == IntentState::Error {
+                        log::warn!(
+                            "not retrying intent ID {}. since it is in state Error",
+                            intent.id,
+                        );
+                        return Err(last_err.unwrap_or(GroupError::Generic(
```

Comment:
I personally would prefer errors like this to be made into concrete enu mvariants like `GroupError::CouldNotBeCommitted` with the message as an `#[error("Group Intent could not be permitted")]` rather than throwing everything into the `Generic` bucket. Should make these errors easy to match on should we need to, + can use the `Retryable` trait on them (in this case no but for other errors)

Reply:
@insipx I agree, I added a comment on the follow up issue here https://github.com/xmtp/libxmtp/issues/822#issuecomment-2153361265.

Think it's better to update this error message while looking at that issue so that this category of errors are more cohesive/ better organized. Thanks for the idea though, I'll make sure this gets updated 🙏